### PR TITLE
Mice now retain custom names on death.  ( Issue #18786 )

### PR DESCRIPTION
**Pull request by SnipeDragon**
This is a fix for the reported consistency issue #18786

### CODE REVIEW

File: code/modules/mob/living/simple_animal/friendly/mouse.dm [hunk 1]
```diff
@@ -44,6 +44,9 @@
 		..(1)
 		var/obj/item/trash/deadmouse/M = new(src.loc)
 		M.icon_state = icon_dead
+		if(src.name != initial(src.name))
+			M.name = src.name
```
Comment (Jordie0608) [inline code/modules/mob/living/simple_animal/friendly/mouse.dm]: All these `src`s are implicit and can be removed

File: code/modules/mob/living/simple_animal/friendly/mouse.dm [hunk 2]
```diff
@@ -44,6 +44,9 @@
...
+			M.desc = "A dead mouse"
```
Comment (Jordie0608) [inline code/modules/mob/living/simple_animal/friendly/mouse.dm]: The dead mouse already has a description and this doesn't add anything to it

**Comment by Bawhoppen** [comment]
you PRed from your master, this will bite you later on

**Comment by SnipeDragon** [comment]
There, pushed the fix with expected coding standards.  Sorry about that.  As for the description, I changed it only because the base description, "It looks like somebody dropped the bass on it." doesn't exactly make sense unless the name is dead mouse.  But I went ahead and removed it.

**Comment by Bawhoppen** [comment]
I'm sure they'd figure out the joke

**Comment by SnipeDragon** [comment]
I'm sure, it was just a result of me over thinking.

**Comment by MrStonedOne** [comment]
Could you give a more descriptive title to your pr, either something like "Mice now retain custom names on death" or "Fixes #18786 (mice losing custom name on death)"

Somebody shouldn't need to visit the issue you fixed to even get a gist of what the pr does.

Also, you should ***always*** create a new branch (based off of master (new branches are normally based off the branch you are viewing when you create it)) for every pr you create.

If this pr got rejected (it's not likely in this case, but still) you would have a dirty master, causing these changes to show up in every other pr you make to us.

Also, you can't make any new prs now until this pr is merged as otherwise both changes will show up in both prs (the new pr's changes would show up here, the changes here would show up in the new pr)

**Comment by MrStonedOne** [comment]
Also, if you mention in the pr description an issue using the #number syntax (like you already do, thanks), and add `fixes` before it, that issue report will auto close when the pr is merged.

ie:
`fixes #18836`

On that note, in the future you don't need to go to the issue and comment that the pr fixes it (you can if you want, you just don't need to), as referring to an issue adds a little note on the issue about the mention, and if you do the fixes bit above, it even has a icon specifying that the referring pr fixes the issue.

(also, don't take my nagging for discouragement, I'm always a bit nitpicky)

**Comment by SnipeDragon** [comment]
Thank you for the advice.  I am mostly familiar with Git from small projects and very small teams (2-3 people).  In fact, I've not really ever pushed commits to public repositories that I couldn't personally merge requests into.

One note though, I can create another pull request by creating a branch based off an older (pre-my-change) commit.  It isn't the cleanest way to go about things, but it does work.

File: code/modules/mob/living/simple_animal/friendly/mouse.dm [hunk 3]
```diff
@@ -44,6 +44,8 @@
 		..(1)
 		var/obj/item/trash/deadmouse/M = new(src.loc)
 		M.icon_state = icon_dead
+		if(name != initial(name))
```
Comment (xxalpha) [inline code/modules/mob/living/simple_animal/friendly/mouse.dm]: I don't think this fixes it. Map edited mice will have their initial() name different than the coded name, so those mice, such as Tom, will generate a 'mouse' on death. Just ditch the if check and always set the 'M.name'.

**Comment by SnipeDragon** [comment]
I removed the if check and just made the dead mouse item take the name of the dying mouse.

**Comment by optimumtact** [comment]
Thanks

**Pull request merged**